NEXT.md

```diff
@@ -0,0 +1,112 @@
+
+
+Including `dataset` as a coordination type allows:
+- different components to be associated with different datasets
+    - facilitates multi-dataset comparisons
+- changing multiple components from some dataset `D1` to some other dataset `D2` would simply require setting one coordination scope value, from `D1` to `D2`.
+    - For example, in the HuBMAP portal, the current dataset dropdown menu could be implemented as a very simple component which just changes the value of the `dataset` coordination object.
+- getting rid of a global `RESET` event for resetting datasets, since components are already listening for changes to coordination objects
+
+
+## Future UI ideas
+
+### Coordination scope dropdowns
+
+As a first step, linking of views will mainly be controlled via view config definitions.
+However, in the future, we would like to implement an interface to allow the end user to update linked views and coordination scopes during the exploration process.
+
+A basic UI would allow the user to view and update the mappings from components to "coordination scopes" via simple dropdowns for each view and coordination type.
+
+<img width="1417" alt="Screen Shot 2020-08-09 at 11 20 17 PM" src="https://user-images.githubusercontent.com/7525285/89749769-242bf980-da97-11ea-8136-4309b70f98c8.png">
+
+<!--
+For example, in the above mockup, notice that the PCA and t-SNE scatterplots have different coordination scopes for the "Cell Color Encoding" coordination type. Perhaps the value of the `Cell Color Encoding` coordination object looks like `{ A: 'cell-set-colors', B: 'gene-expression' }`, meaning that the PCA scatterplot cells would be colored by the colors of the currently-selected cell sets, while the t-SNE scatterplot cells would be colored by the gene expression values of the currently-selected gene.
+
+This would also facilitate multi-dataset comparisons. For example, if you want to compare `linnarsson-2018` to `dries-2019`, just create two different scopes for the `dataset` coordination object. Then, map the half of the components to scope `A` and the other half to scope `B`, where the coordination object value looks like  `{ A: 'linnarsson-2018', B: 'dries-2019' }`. Then, let's say you want to switch the comparison, and now you want to compare `linnarsson-2018` to `spraggins-2020`. Then the only thing you need to do is update the coordination object `{ A: 'linnarsson-2018', B: 'spraggins-2020' }`. (Assuming the view config `datasets` property was an array containing the file URL mappings for all 3 datasets `linnarsson-2018`, `dries-2019`, and `spraggins-2020`).
+-->
+
+### Coordination object graph representation
+
+One may think of the coordination objects, components, and scopes as forming a network or graph. It may be helpful to display this graph to the user.
```

Comment:
This would be a bi-partite graph, with coordinations linked only to views, and views linked only to coordinations? Matrix representation, with coordinations on one axis and views on the other, might work.

Reply:
Right, so for a single coordination type the graph would look like this:

![Coordination model as graph representation - single coordination type](https://user-images.githubusercontent.com/7525285/92250949-1ef57b00-ee9a-11ea-80e3-9b5b668de38e.png)

<hr>
Considering more than one coordination type at a time may start to get messy. In the figure below, I have encoded the coordination type with a shape. Alternatively, you could imagine an additional column in the graph (tri-partite?) which links coordination types to coordination scopes. (For the below example, that column would contain two nodes, one for "rotation" and another for "zoom")

![Coordination model as graph representation - multiple coordination types](https://user-images.githubusercontent.com/7525285/92250958-2157d500-ee9a-11ea-90b3-f5746adef911.png)

Reply:
That extra column / tri-partite representation would look like
![Coordination model as graph representation - multiple coordination types - node encoding](https://user-images.githubusercontent.com/7525285/92252149-d6d75800-ee9b-11ea-8f82-11936e4f5076.png)